modules not working correctly

### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain.js documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain.js rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

import { PremEmbeddings } from "@langchain/community/embeddings/premai";
import { Pinecone } from '@pinecone-database/pinecone';
import { PineconeStore } from "@langchain/pinecone";


export async function getVectorStore(){
    const embeddings = new PremEmbeddings({
        // In Node.js defaults to process.env.PREM_API_KEY
        apiKey: process.env.PREM_API_KEY,
        // In Node.js defaults to process.env.PREM_PROJECT_ID
        project_id: 1744,
        model: "text-embedding-3-large", // The model to generate the embeddings
      });
      
    const pineconeClient = new Pinecone({
        apiKey: process.env.PINECONE_API_KEY!
    });

    const index = pineconeClient.Index(process.env.PINECONE_INDEX_NAME!);

    const vectorStore = await PineconeStore.fromExistingIndex(embeddings, {
        pineconeIndex: index,
        textKey: 'text',
      });
    
    return vectorStore
}




### Error Message and Stack Trace (if applicable)

Package path ./structured_query is not exported from package /Users/prathamchauhan/Documents/Next-Projects/pratham-portfolio/node_modules/@langchain/core (see exports field in /Users/prathamchauhan/Documents/Next-Projects/pratham-portfolio/node_modules/@langchain/core/package.json)

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/@langchain/pinecone/dist/index.js
./node_modules/@langchain/pinecone/index.js
./src/lib/pineconedb.ts
./src/actions/chat.ts
./src/app/layout.tsx

### Description

I am trying to get the pinecone db vector store to query the data that I saved by running a different ts script.
Not only this it is giving me error with neon vector store 
Cannot find module '@langchain/community/vectorstores/neon' or its corresponding type declarations.
Even though I have gone through docs and github discussions

### System Info

npm version 10.7.0

--- a/libs/langchain-pinecone/package.json
+++ b/libs/langchain-pinecone/package.json
@@ -39,7 +39,7 @@
   "author": "Pinecone, Inc",
   "license": "MIT",
   "dependencies": {
-    "@langchain/core": ">0.1.0 <0.3.0",
+    "@langchain/core": ">0.2.0 <0.3.0",
     "@pinecone-database/pinecone": "^2.2.0",
     "flat": "^5.0.2",
     "uuid": "^9.0.0"

--- a/libs/langchain-weaviate/package.json
+++ b/libs/langchain-weaviate/package.json
@@ -40,7 +40,7 @@
   "author": "LangChain",
   "license": "MIT",
   "dependencies": {
-    "@langchain/core": ">0.1.0 <0.3.0",
+    "@langchain/core": ">0.2.0 <0.3.0",
     "uuid": "^9.0.1",
     "weaviate-ts-client": "^2.0.0"
   },

